libtransmission/session.h

```diff
@@ -404,6 +405,8 @@ struct tr_session
     // Only session.cc should use this.
     int peer_socket_tos_ = *tr_netTosFromName(TR_DEFAULT_PEER_SOCKET_TOS_STR);
 
+    tr_bandwidth_group* groups;
```

Comment:
let's use a std::map here so that the linked list can be removed, i.e. to decouple the `tr_bandwidth_group` struct from how `tr_session` manages a collection of them

`std::map<std::string,tr_bandwidth_group> bandwidth_groups;`